namesys/ipns_publisher.go

```diff
@@ -66,9 +64,9 @@ func IpnsDsKey(id peer.ID) ds.Key {
 //
 // This method will not search the routing system for records published by other
 // nodes.
-func (p *IpnsPublisher) ListPublished(ctx context.Context) (map[peer.ID]*ipns.Record, error) {
+func (p *IPNSPublisher) ListPublished(ctx context.Context) (map[peer.ID]*ipns.Record, error) {
```

Comment:
This code isn't used anywhere. It's been sitting here for 5 years without being used. Can I kill it?

Reply:
fwiw it is mentioned in:  https://github.com/ipfs/boxo/blob/25ae0bd151b43ff3ec56aac44fc8a03337a72f7b/namesys/republisher/repub.go#L98   

I would avoid making waves and keep it, but  use `ipns.Name` like you suggested

Reply:
@hacdias ack, i guess the best thing we have for this is CID from go-cid? (with libp2p-key codec)
Still feels better than leaking libp2p PeerID into namesys interface.

Reply:
@lidel I changed `ipns.Name.src` from `[]byte` to `string` and `ipns.Name` is now comparable. I think that is perfectly fine as that variable is only used internally.